<launch>
    <!-- Detection configuration -->
    <arg name="weights" default="/home/kemove/delta_project/detection/catkin_yolo/src/yolo_detect/src/yolo11n.engine" />
    <arg name="confidence_threshold" default="0.4" />
    <arg name="iou_threshold" default="0.45" />
    <arg name="classes" default="[0,1,2]" />
    <arg name="maximum_detections" default="1000" />
    <arg name="agnostic_nms" default="true" />
    <arg name="line_thickness" default="2" />
    <arg name="inference_size_h" default="480" />
    <arg name="inference_size_w" default="480" />
    <arg name="input_image_topic" default="/zed2i/zed_node/left/image_rect_color" />
    <arg name="output_topic" default="/yolov5/detections" />
    <arg name="publish_image" default="true" />
    <arg name="output_image_topic" default="/yolov5/image_out" />


    <node pkg="yolo_detect" name="detection" type="detection.py" output="screen">
        <param name="weights" value="$(arg weights)" />
        <param name="classes" value="$(arg classes)" />
        <param name="confidence_threshold" value="$(arg confidence_threshold)" />
        <param name="iou_threshold" value="$(arg iou_threshold)" />
        <param name="maximum_detections" value="$(arg maximum_detections)" />
        <param name="agnostic_nms" value="$(arg agnostic_nms)" />
        <param name="line_thickness" value="$(arg line_thickness)" />

        <param name="inference_size_h" value="$(arg inference_size_h)" />
        <param name="inference_size_w" value="$(arg inference_size_w)" />

        <param name="input_image_topic" value="$(arg input_image_topic)" />
        <param name="output_topic" value="$(arg output_topic)" />


        <param name="publish_image" value="$(arg publish_image)" />
        <param name="output_image_topic" value="$(arg output_image_topic)" />
    </node>

</launch>
